pandas/tests/frame/methods/test_combine_first.py

```diff
@@ -339,9 +343,13 @@ def test_combine_first_int(self):
         df1 = pd.DataFrame({"a": [0, 1, 3, 5]}, dtype="int64")
         df2 = pd.DataFrame({"a": [1, 4]}, dtype="int64")
 
-        res = df1.combine_first(df2)
-        tm.assert_frame_equal(res, df1)
-        assert res["a"].dtype == "int64"
+        res1 = df1.combine_first(df2)
+        res2 = df1.combine_first(df2)
+        # this would fail prior to this fix
+        assert res1["a"].dtype == res2["a"].dtype
```

Comment:
same comments

Reply:
the `dtype` of the results are different in master

```python
import pandas as pd

df1 = pd.DataFrame({"a": [0, 1, 3, 5]}, dtype="int64")
df2 = pd.DataFrame({"a": [1, 4]}, dtype="int64")

res1 = df1.combine_first(df2)
res2 = df2.combine_first(df1)

print(res1["a"].dtype, res2["a"].dtype)
```